packages/pigeon/lib/objc_generator.dart

```diff
@@ -420,6 +458,26 @@ class ObjcSourceGenerator extends StructuredGenerator<ObjcOptions> {
     indent.newln();
   }
 
+  @override
+  void writeEnum(
+    ObjcOptions generatorOptions,
+    Root root,
+    Indent indent,
+    Enum anEnum, {
+    required String dartPackageName,
+  }) {
+    final String enumName = _className(generatorOptions.prefix, anEnum.name);
+    addDocumentationComments(
+        indent, anEnum.documentationComments, _docCommentSpec);
+
+    indent.writeln('@interface ${enumName}Wrapper ()');
+    indent.writeln('@end');
+    indent.newln();
+    indent.writeln('@implementation ${enumName}Wrapper');
+    indent.writeln('@end');
```

Comment:
We don't follow the `if (self)` pattern anywhere else in this package, is it necessary here?

Reply:
It is always best practice to do some form of the check (another variant is `if ((self = [super init])) {`), because `init` is not guaranteed to return a non-nil value.

Are we ever going to end up in a situation where that happens in a class that inherits from `NSObject`? Almost certainly not, but I'm not aware of any argument for **not** doing it, and consistently applying the pattern means not forgetting it in cases where it actually could be necessary in practice.